flowkit/arguments/arguments.go

```diff
@@ -145,14 +147,21 @@ func ParseWithoutType(args []string, code []byte, fileName string) (scriptArgs [
 			break
 		}
 
-		var value, err = runtime.ParseLiteral(argumentString, semaType, inter)
```

Comment:
Maybe instead/also report this upstream in https://github.com/onflow/cadence.
`ParseLiteral` should support `nil` and optional types, I'm surprised it does not at the moment.

Reply:
From what I can see, Cadence already supports parsing nil/optionals, here's a test case  for it: https://github.com/onflow/cadence/blob/5c53da527bdd3b814382bb9c87cd99e5351616e4/runtime/literal_test.go#L69-L80. What is being passed as arguments to `ParseLiteral` that makes it return an error?

Reply:
I think sematype check is erroring then, let me try to fix that way

Reply:
The bug is likely the unwrapping of the optional type here: https://github.com/onflow/flow-cli/pull/996/files#diff-116113b82ac2b212970f78016025c6aa67609801d3ae0d98aa3cc00fafbb7e29R131

Reply:
fixed in https://github.com/onflow/flow-cli/pull/996/commits/83a5d514508c8623a655291f5477978a60c62a5c

Reply:
@turbolent hmm how can I check if it is Address or String, I need for that part only to unwrap. Actually commit from you would be the best if not much hassle

Reply:
Refactored the function and improved the tests in c3a5c074293aa93c9f5f4d7d14898e92fbeb4722